### PR TITLE
Make sticky side navigation optional

### LINKED ISSUE #3206: Option to remove scrollbar in side-nav
## Pattern to amend

Requested option to remove scrollbar on Side Navigation, when the content exceeds the height of the container

## Visual
Screenshot from 2020-08-14 16-58-04

## Context

This request came from Anbox-cloud to remove scrollbar on the left side of the navigation. This could easily be done I suppose by adding a modifier to opt out of the sticky position

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Removes sticky functionality from default side navigation.
Adds `is-sticky` modifier class to enable sticky side navigation on large screens.

Fixes #3206

## QA

- Run `./run` or [demo]( https://vanilla-framework-3427.demos.haus/docs)
- Check the examples of side navigation (both desktop and mobile view) and make sure they work as expected:
  - [default](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/docs) and [with icons](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/icons) or [raw HTML](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/raw-html) - should not be sticky on large screen, mobile drawer should work as before
  - new [sticky](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/sticky) variant should be sticky when page is scrolled. bigger example in [docs layout](https://vanilla-framework-3427.demos.haus/docs/examples/layouts/documentation_
  - verify also comparison examples [light](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/default) and [dark](https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/dark) - only sticky nav (3rd) should be sticky
- Check updated [side navigation docs](https://vanilla-framework-3427.demos.haus/docs/patterns/navigation#sticky)

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
@bartaz flipping through the examples this has a margin-left on small screen and the others don't - why? https://vanilla-framework-3427.demos.haus/docs/examples/patterns/side-navigation/raw-html

**Review by sowasred2012 — APPROVED** [review]
Looks good, just one question and one small grammar tweak.

**Comment by bartaz** [comment]
@lyubomir-popov 

Good catch. As far as I can see it's because "raw html" example includes "row" as a wrapper, while some other examples don't. Will make them consistent.

File: templates/docs/examples/patterns/side-navigation/sticky.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% extends "_layouts/examples.html" %}
+{% block title %}Side navigation / Sticky{% endblock %}
+
+{% block standalone_css %}patterns_side-navigation{% endblock %}
+
+{% block content %}
+{% set is_sticky = True %}
```
[the remark this answers is not in the document]
Comment (bartaz) [reply templates/docs/examples/patterns/side-navigation/sticky.html:7]: @sowasred2012 Good call. I didn't know about `{% with %}`, I will use it, as it seems to give better control on the scope of variables.

**Pull request merged**